sdk/iot/provisioning/inc/az_iot_provisioning_client.h

```diff
@@ -165,15 +187,19 @@ AZ_NODISCARD az_result az_iot_hub_client_sas_password_get(
  * @brief Gets the MQTT topic filter to subscribe to register responses.
  *
  * @param[in] client The #az_iot_provisioning_client to use for this call.
- * @param[in] mqtt_topic_filter An empty #az_span with sufficient capacity to hold the MQTT topic
- *                              filter.
- * @param[out] out_mqtt_topic_filter The output #az_span containing the MQTT topic filter.
+ * @param[out] mqtt_topic_filter A buffer with sufficient capacity to hold the MQTT topic filter.
+ *                               If successful, contains a null-terminated string with the topic
+ *                               filter that needs to be passed to the MQTT client.
+ * @param[in] mqtt_topic_filter_size The size, in bytes of \p mqtt_topic_filter.

```

Comment:
```suggestion
 * @param[in] mqtt_topic_filter_size The maximum available space, in bytes, within the buffer referred to by \p mqtt_topic_filter.
```

Reply:
If you agree, apply the similar wording across the board.

Reply:
The current wording is in-line with az_span which has a ptr and size. Changes to doc wording should be made across the Az SDK. If we decide to document `size` as  "the maximum available space" let's do it across all APIs. Please open a tracking bug for that.